File: agent-web/src/model/Product.js
export class Product {
  id;
  name;
  price;
  onStock;
  imageId;

  constructor(id, name, price, onStock, imageId) {
    this.id = id;
    this.name = name;
    this.price = price;
    this.onStock = onStock;
    this.imageId = imageId;
  }
}

export const newProduct = () => new Product(null, "", 0, 0, 0);

File: agent-report/src/main/java/rs/ac/uns/ftn/devops/tim5/agentreport/model/Product.java
package rs.ac.uns.ftn.devops.tim5.agentreport.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private Double price;

    private Integer onStock;

    private Boolean deleted;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn
    private Image image;
}
